### ISSUE TITLE
Incorrect dependency between aws_eks_cluster and aws_eks_fargate_profile

### ISSUE THREAD

**Issue by RomanKrasavtsev** [hashicorp/terraform-provider-aws#12179]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.12.21
+ provider.aws v2.50.0

### Affected Resource(s)

* aws_eks_cluster
* aws_eks_fargate_profile

### Expected Behavior
After tainting EKS cluster, it should be recreated along with Farget profile

### Actual Behavior
```
Plan: 1 to add, 0 to change, 1 to destroy.
...
aws_eks_cluster.cluster: Destroying... [id=test-eks]

Error: error deleting EKS Cluster (test-eks): ResourceInUseException: Cluster has Fargate Profiles in ACTIVE/CREATING state
{
  ClusterName: "test-eks",
  Message_: "Cluster has Fargate Profiles in ACTIVE/CREATING state"
}
```
### Steps to Reproduce

0. Create EKS cluster and Fargate profile.
1. `terraform taint aws_eks_cluster.cluster`
2. `terraform apply`

**Comment by hryamzik** [comment]
I guess the only work-around is to do full destroy-apply each time cluster requires recreation. To avoid similar errors on destroy make fargate profiles depend on each other and on cluster resource. (profile A depends on cluster, profile B depends on profile A, etc).

**Comment by bflad** [comment]
Hi folks 👋 

This error message, when caused by concurrent deletions of EKS Fargate Profiles, was recently addressed in [version 3.16.0 of the Terraform AWS Provider](https://github.com/hashicorp/terraform-provider-aws/blob/master/CHANGELOG.md#3160-november-18-2020) by serializing those requests with creating/deleting EKS Fargate Profiles of the same EKS Cluster.

The other potential issue with tainting a Terraform AWS Provider `aws_eks_cluster` resource and not having downstream `aws_eks_node_group` resources recreated however, is not a behavior that we have control over in the Terraform AWS Provider codebase. The logic for handling the ordering and behaviors of the `taint` command lives upstream in the Terraform CLI codebase. If you are having issues with particular parts of its behavior, an issue will need to be filed in that repository at: https://github.com/hashicorp/terraform/issues
